### PR TITLE
Convert BigInteger to Int64 when possible from Python

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose

https://autodesk.slack.com/archives/CED3BF6C9/p1571942488031200
image

JIRA: https://jira.autodesk.com/browse/DYN-2262

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@DynamoDS/dynamo 

### FYIs

@Dewb

### CODE REVIEW

File: src/DynamoUtilities/DataMarshaler.cs [hunk 1]
```diff
@@ -86,7 +87,20 @@ public object Marshal(object obj)
             var applicable = marshalers.Where(pair => pair.Key.IsAssignableFrom(targetType));
 
             if (!applicable.Any())
-                return obj;
+            {
+                if (!(obj is BigInteger bigInt)) return obj;
```
Comment (aparajit-pratap) [inline src/DynamoUtilities/DataMarshaler.cs]: C# 7.0 syntax seems to be breaking the engops build.

**Comment by Dewb** [comment]
Excellent! 

Any reason this has to be done inside the generic marshaller execution rather than by defining a custom marshaler for `BigInteger`?

**Comment by Dewb** [comment]
Looks like we have custom `DataMarshaler`s for Python node inputs, but not outputs. This new code could definitely go here: https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/DSIronPython/IronPythonEvaluator.cs

The `DataMarshaler` class isn't used anywhere aside from the Python node so I would also support moving it out of `DynamoUtilities` and into the Python project, to make its purpose more clear. (Doesn't have to be part of this PR.)

**Comment by aparajit-pratap** [comment]
> Looks like we have custom `DataMarshaler`s for Python node inputs, but not outputs. This new code could definitely go here: https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/DSIronPython/IronPythonEvaluator.cs

The `DataMarshaler` class is used for both input and output marshaling as seen here: https://github.com/DynamoDS/Dynamo/blob/8fb3b6829c9bcd8c01ee4ef7e880aa8185bf2cd0/src/Libraries/DSIronPython/IronPythonEvaluator.cs#L114 and here: https://github.com/DynamoDS/Dynamo/blob/8fb3b6829c9bcd8c01ee4ef7e880aa8185bf2cd0/src/Libraries/DSIronPython/IronPythonEvaluator.cs#L134, however, I agree that the marshaling for `BigInteger` can be specific to the output marshaling.

**Comment by mjkkirschner** [comment]
@aparajit-pratap would you mind filing this internally?

**Comment by aparajit-pratap** [comment]
Added unit test, addressed review comments about registering the `BigInteger` marshaling logic only for node output (output marshalers are added only in the `DataMarshaler` constructor, while input marshalers are added when first initializing the `IronPythonEvaluator` property). 

I haven't moved `DataMarshaler` to the Python project just yet as @mjkkirschner had some other thoughts on this. Leaving it open for discussion.

File: src/DynamoUtilities/DynamoUtilities.csproj [hunk 2]
```diff
@@ -49,6 +49,7 @@
     <Reference Include="System" />
     <Reference Include="System.configuration" />
     <Reference Include="System.Core" />
+    <Reference Include="System.Numerics" />
```
Comment (QilongTang) [inline src/DynamoUtilities/DynamoUtilities.csproj]: Do we need this reference?

**Comment by QilongTang** [comment]
LGTM with one comments

File: src/DynamoUtilities/DynamoUtilities.csproj (hunk 2 above)
Comment (aparajit-pratap) [reply src/DynamoUtilities/DynamoUtilities.csproj]: Yes, `BigInteger` is defined in `System.Numerics`.

**Comment by Dewb** [comment]
FYI the IronPython issue that prompted this is supposedly fixed in IronPython 2.7.9.

**Comment by mjkkirschner** [comment]
hmm - so this test should have passed already - since we are using ironPython 2.7.9.

@aparajit-pratap do we now include system.numerics in our shipped binaries or this is loaded from the GAC?

It will be useful to know as we should communicate this in the release notes of 2.5

**Comment by aparajit-pratap** [comment]
I reverted my change leaving the test to test locally. It didn't pass?

**Comment by mjkkirschner** [comment]
@aparajit-pratap your test also includes a big integer check (last type in your list) - perhaps that gets marshalled to long now in all cases?

**Comment by mjkkirschner** [comment]
here's Mike's issue:
https://github.com/IronLanguages/ironpython2/issues/681

**Comment by aparajit-pratap** [comment]
> @aparajit-pratap your test also includes a big integer check (last type in your list) - perhaps that gets marshalled to long now in all cases?

This is the output I'm seeing before my change:
image

**Comment by aparajit-pratap** [comment]
> @aparajit-pratap do we now include system.numerics in our shipped binaries or this is loaded from the GAC?

It is loaded from GAC and referenced from `C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7\`

**Pull request merged**